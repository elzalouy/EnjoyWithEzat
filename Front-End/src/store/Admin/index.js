import { createSlice } from "@reduxjs/toolkit";
import initialState from "./initialState";
import reducers from './reducers';
const AdminSlice = createSlice({
  name: "admin",
  initialState,
  reducers: reducers
});
export const AdminActions=AdminSlice.actions;
export default AdminSlice.reducer;
