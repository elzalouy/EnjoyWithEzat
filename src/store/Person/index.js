import { createSlice } from "@reduxjs/toolkit";
import initialState from "./initialState";

const PersonSlice = createSlice({
  name: "person",
  initialState,
  reducers: {
    selectProjectsByType(state,action){
    }
  },
});
export const PersonActions = PersonSlice.actions;
export default PersonSlice.reducer;
