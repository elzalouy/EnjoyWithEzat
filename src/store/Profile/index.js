import { createSlice } from "@reduxjs/toolkit";
import initialState from "./initialState";

const ProfileSlice = createSlice({
  name: "person",
  initialState,
  reducers: {
    setLanguagePercent(state,action){
      state.languages.find(item=>item.key===action.payload.element).value=action.payload.value;
    }
  },
});
export const PersonActions = ProfileSlice.actions;
export default ProfileSlice.reducer;
