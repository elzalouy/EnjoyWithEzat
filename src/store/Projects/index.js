import { createSlice } from "@reduxjs/toolkit";
import initialState from "./initialState";
import reducers from "./reducers";

const ProjectsSlice = createSlice({
  name: "projects",
  initialState,
  reducers:reducers
});

export const ProjectsActions = ProjectsSlice.actions;
export default ProjectsSlice.reducer;