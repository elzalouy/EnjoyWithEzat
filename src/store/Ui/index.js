import { createSlice } from "@reduxjs/toolkit";
import UiinitialState from "./initailState";
import reducers from './reducers';


const UiSlice = createSlice({
  name: "UIs",
  initialState: UiinitialState,
  reducers:reducers
});

export const UiActions = UiSlice.actions;
export default UiSlice.reducer;