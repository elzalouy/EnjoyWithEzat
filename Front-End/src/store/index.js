import { configureStore } from "@reduxjs/toolkit";
import UiSlice from "./Ui/index";
import PersonSlice from "./Profile/index";
import Projects from './Projects/index'
import Admin from "./Admin/index";
const store = configureStore({
  reducer: { UI: UiSlice, Person: PersonSlice, Projects: Projects, Admin : Admin },
});
export default store;