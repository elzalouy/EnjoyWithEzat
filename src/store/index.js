import { configureStore } from "@reduxjs/toolkit";
import UiSlice from "./Ui/index";
import PersonSlice from "./Person/index";
import Projects from './Projects/index'
const store = configureStore({
  reducer: { UI: UiSlice, Person: PersonSlice, Projects: Projects},
});
export default store;
