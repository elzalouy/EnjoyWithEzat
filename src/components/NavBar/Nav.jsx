import React ,{ Fragment } from "react";
import { useDispatch, useSelector } from "react-redux";
import { UiActions } from "../../store/Ui";
import classes from "./Nav.module.css";

const Nav= () => { 
  const dispatch = useDispatch();
  // Switching between sections
  const sectionToggleHandler = (e, name) => {
    dispatch(UiActions.Switch(name));
  };

  //getting the sections
  const { sections, curruntSection } = useSelector((state) => state.UI);
  const style =
    curruntSection.color === "gold" ? classes.navBlack : classes.navGold;
  return (
    <Fragment>
      <ul className={classes.nav + " Shadow " + style}>
        {sections && 
          sections.map((section) => (
            <li
              key={section.key}
              className={
                classes.navItem +
                (section.key === curruntSection.key ? " active" : "")
              }
              onClick={(e) => sectionToggleHandler(e, section.name)}
            >
              <i className={section.icon}></i>
            </li>
          ))}
      </ul>
    </Fragment>
  );
};
export default Nav;