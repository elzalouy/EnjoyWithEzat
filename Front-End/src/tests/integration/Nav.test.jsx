import Nav from '../../components/NavBar/Nav';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import createMockStore from 'redux-mock-store';
import reduxStore from '../../store';
import { UiActions } from '../../store/Ui';
const mockstore=createMockStore([]);
describe('Nav component:-', () => {
    let store=mockstore({UI:reduxStore.getState().UI,Person:reduxStore.getState().Person});
     store.dispatch=jest.fn();
    let component=renderer.create(<Provider store={store}><Nav/></Provider>);
    it('component should be rendered:-',()=>{
        expect(component.toJSON()).toMatchSnapshot();
    });
    it('should swicth to platform component:-',()=>{
        renderer.act(()=>{
            component.root.findAllByType('li')[1].props.onClick();
        });
        expect(store.dispatch).toBeCalledTimes(1);
        expect(store.dispatch).toBeCalledWith(UiActions.Switch('platform'));
    })
})
