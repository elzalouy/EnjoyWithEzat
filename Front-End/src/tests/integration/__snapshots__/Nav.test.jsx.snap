// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component:- component should be rendered:- 1`] = `
<ul
  className="nav Shadow navBlack"
>
  <li
    className="navItem active"
    onClick={[Function]}
  >
    <i
      className="bi bi-person"
    />
  </li>
  <li
    className="navItem"
    onClick={[Function]}
  >
    <i
      className="bi bi-file-earmark-richtext"
    />
  </li>
  <li
    className="navItem"
    onClick={[Function]}
  >
    <i
      className="fas fa-tasks"
    />
  </li>
  <li
    className="navItem"
    onClick={[Function]}
  >
    <i
      className="fas fa-award"
    />
  </li>
</ul>
`;
